spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySourceLocator.java

```diff
@@ -176,22 +181,28 @@ private String propWithProfile(String context, String profile) {
      * When generating more than one it needs to be in the last one.
      * @return a list of AzureConfigPropertySources
      * @throws IOException
+     * @throws URISyntaxException
      */
     private List<AzureConfigPropertySource> create(String context, ConfigStore store,
-            Map<String, List<String>> storeContextsMap, boolean initFeatures, FeatureSet featureSet)
-            throws IOException {
+            Map<String, List<String>> storeContextsMap, boolean initFeatures, FeatureSet featureSet) throws Exception {
         List<AzureConfigPropertySource> sourceList = new ArrayList<>();
 
-        for (String label : store.getLabels()) {
-            AzureConfigPropertySource propertySource = new AzureConfigPropertySource(context, operations,
-                    store.getName(), label, properties, appProperties);
+        try {
+            for (String label : store.getLabels()) {
+                AzureConfigPropertySource propertySource = new AzureConfigPropertySource(context, store.getName(),
+                        label,
+                        properties, appProperties, clients);
 
-            propertySource.initProperties(featureSet);
-            if (initFeatures) {
-                propertySource.initFeatures(featureSet);
+                propertySource.initProperties(featureSet);
+                if (initFeatures) {
+                    propertySource.initFeatures(featureSet);
+                }
+                sourceList.add(propertySource);
+                putStoreContext(store.getName(), context, storeContextsMap);
             }
-            sourceList.add(propertySource);
-            putStoreContext(store.getName(), context, storeContextsMap);
+        } catch (Exception e) {
+            delayException();
```

Comment:
What's the purpose of delayException?

Reply:
There was a thing seen in applications that automatically restart when they fail. In these cases a large number of requests to App Configuration can be made in a very short period of time. The idea behind delayException is to delay a restart if App Configuration fails to load for any reason, this is done by setting a minimum up time before any error is thrown.